packages/react/src/Card/Card.client.tsx

```diff
@@ -0,0 +1,37 @@
+import { useMemo, type ComponentProps } from "react";
+import { getComponentClassName } from "../Form/core";
+
+type CardProps = ComponentProps<"div"> &
+  ComponentProps<"button"> & {
+    classModifier?: string;
+  };
+
+export const Card = ({
+  children,
+  className,
+  classModifier,
+  onClick,
+  ...otherProps
+}: CardProps) => {
+  const componentClassName = useMemo(
+    () => getComponentClassName(className, classModifier, "af-card"),
+    [className, classModifier],
+  );
+
+  return onClick ? ( // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+    <button
+      type="button"
+      className={componentClassName}
+      tabIndex={0}
+      onClick={onClick}
+      {...otherProps}
+    >
+      {children}
+    </button>
+  ) : (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+    <section className={componentClassName} tabIndex={0} {...otherProps}>
+      {children}
+    </section>
+  );
```

Comment:
Juste car je trouve que c'est plus propre, 

```suggestion
  const Component = onClick ? 'button' ? 'section';
  const ComponentProps = Component === 'button'
    ? {
    type: 'button", //... mettre toute les props
    }
    : {
     // ... idem ici
    }
    
  // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
  return <Component {...ComponentProps}>{children}</Component>;
```

Reply:
J'ai comme résultat ça :
![image](https://github.com/AxaFrance/design-system/assets/81412571/1fbabde4-3a8e-45ed-aaf2-ed8047a2ce52)

Ce sera assez lisible comme code ?

Reply:
Niquel, ha un useMemo bien utilisé sa fait sisir ...